server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4758,11 +4767,21 @@ private String copySegmentFiles(
                 }
             }
             for (String file : uploadedSegments.keySet()) {
```

Comment:
we should also record the overall time for downloading all the new segments. A moving avg of this time would help understand how much time it takes for a replica to sync

Reply:
The set of metrics being published in download stats has been finalized already for this iteration. I will follow this PR up with some new issues to add more metrics and will include this in that.